lib/ansible/modules/cloud/amazon/cloudwatchlogs_log_group.py

```diff
@@ -0,0 +1,149 @@
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- cloudwatchlogs_log_group:
+    log_group_name: test-log-group
+
+- cloudwatchlogs_log_group:
+    state: present
+    log_group_name: test-log-group
+    tags: { "Name": "test-log-group", "Env" : "QA" }
+
+- cloudwatchlogs_log_group:
+    state: present
+    log_group_name: test-log-group
+    tags: { "Name": "test-log-group", "Env" : "QA" }
+    kms_key_id: arn:aws:kms:region:account-id:key/key-id
+
+- cloudwatchlogs_log_group:
+    state: absent
+    log_group_name: test-log-group
+
+'''
+
+# TODO: Disabled the RETURN as it was breaking docs building. Someone needs to
+# fix this
+RETURN = '''# '''
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import _camel_to_snake, camel_dict_to_snake_dict, boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def create_log_group(client, log_group_name, kms_key_id, tags):
+
+    request = {
+        'logGroupName': log_group_name
+    }
+    if kms_key_id:
+        request['kmsKeyId'] = kms_key_id
+    if tags:
+        request['tags'] = tags
+
+    client.create_log_group(**request)
+
+
+def delete_log_group(client, log_group_name):
+    desc_log_group = client.describe_log_groups(logGroupNamePrefix=log_group_name)
+
+    if 'logGroups' in desc_log_group:
+        for i in desc_log_group['logGroups']:
+            if log_group_name in i['logGroupName']:
+                client.delete_log_group(logGroupName=log_group_name)
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        log_group_name=dict(required=True, type='str'),
+        state=dict(choices=['present', 'absent'],
+                   default='present'),
+        kms_key_id=dict(required=False, type='str'),
+        tags=dict(required=False, type='dict')
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required.')
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        if not region:
+            module.fail_json(msg="Region must be specified as a parameter, in EC2_REGION or AWS_REGION environment variables or in boto configuration file")
+        logs = boto3_conn(module, conn_type='client', resource='logs', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except botocore.exceptions.ProfileNotFound as e:
+        module.fail_json(msg=str(e))
+
+    state = module.params.get('state')
+
+    if state == 'present':
+        create_log_group(client=logs, log_group_name=module.params['log_group_name'], kms_key_id=module.params['kms_key_id'], tags=module.params['tags'])
```

Comment:
What happens if the log group already exists?

What happens if someone wants to change the tags of the logs group?

Reply:
The ResourceAlreadyExistsException was raised, if the log group already existed.
But I changed this to return the describe_logs information if the log group already exist, and the execution do not failure and ensure idempotency.

I put more two funcionalities.:
a)  overwrite, if the user wants override the log group.
b)  state = get . If the user wants list log group.

I guess to handle log group tags , should be another module.
Maybe a cloudwatchlogs_tag with these funcionalities.:
a) tag_log_group()  = state present
b) untag_log_group() = state absent
c) list_tags_log_group() = state get

If you agree, I will write this module